devops/scripts/deploy_shared_service.sh

```diff
@@ -3,17 +3,18 @@
 set -o errexit
 set -o pipefail
 # Uncomment this line to see each command for debugging (careful: this will show secrets!)
-# set -o xtrace
+set -o xtrace
 
 function usage() {
     cat <<USAGE
 
-    Usage: $0 [-u --tre_url]  [-c --current] [-i --insecure]
+    Usage: $0 [-u --tre_url]  [-c --current] [-i --insecure] [-p --properties]
```

Comment:
The properties thing feels a little "fragile"

What gets passed in are name value pairs that *must* use confirm to a particular format (i.e. be a snippet of JSON with the right quotes etc).

I can see that it works, but it's not obvious (and easy to get wrong). For example, without inspecting the place you pass properties today, you probably wouldn't have guessed that they need to be...

- double quoted name and value
- name/value need to be separated by colon
- name/value pairs need separating by comma

```
"domain_prefix": "nexus","cert_name": "nexus-ssl"
```

Reply:
Having said that, I don't have an obviously better solution to this.
See what other say, but at present I'm not keen on...

1. the values being stored in the `Makefile`
2. the values being passed as a fragment of JSON that must have very specific formatting
3. the help/usage of this script giving no clues on that format

Reply:
How about this as an approach. I prefer this because...

- it's in keeping with how you would pass multiple things to a bash script
- the only format you need to document is an easier `name=value`

https://gist.github.com/martinpeck/41bc86dec7e470503d3f1cd27f968846

Reply:
I've created two options and illustrated them in this Gist:

https://gist.github.com/martinpeck/41bc86dec7e470503d3f1cd27f968846

I prefer option two, but both feel cleaner and more obvious (and allow for extension if other services need properties)

Reply:
Yeah I do agree, I guess I didn't think much of it as it's only being used for one of the shared services currently. The second approach seems quite clean, have implemented that. Thanks for the gists, they were helpful!